### ISSUE TITLE
All gabc files under scores/gabc/ ?

### ISSUE THREAD

**Issue by martindonat** [gregorio-project/GregoBase#11]
For a bigger chant book-project I would need a lot of gabc scores from Gregobase, so I don't want to download them manually. I know, it shouldn't be such a problem: Download this repo, set up a local MySQL server, run the correct php-script, etc

After some considerable amount of time and effort I have a local MariaDB server, I imported the scores, but now I have encoding problems and am fed up... Sorry, I'm just not so familiar with SQL servers, php and this stuff. So I wanted to ask for this little favour, whether you could provide all gabc files in a dir like scores/gabc/.

I guess for you it would be much easier than for me.

Thanks in advance.

**Comment by igneus** [comment]
Do you still need the scores?

**Comment by martindonat** [comment]
No, thanks. I did it the quick and dirty way: A python script which downloads file per file from the website. It looks like this:

`import urllib.request

def extract_filename(input_str, num):

	filename_pos = input_str.find("filename=")
	
	# "filename=" nicht gefunden
	if filename_pos == -1:
	
		print("Filename not found: {0}".format(num))
		return None
	
	# "filename=" gefunden	
	else:
	
		# vorne abschneiden
		filename = input_str[filename_pos + 9:]
	
		# evtl hinten abschneiden
		if filename.find(";") != -1:
	
			filename = filename[:filename.find(";")]
	
		return filename

for i in range(1148,9000):

	print("Download file ... {0}".format(i))

	# Request to URL
	gabc_download = urllib.request.urlopen("https://gregobase.selapa.net/download.php?id={0}&format=gabc".format(i))
	
	# Get Info Header
	info_str = gabc_download.info()['Content-Disposition']
	
	# Check if anything is found
	if info_str:
	
		#Extract filename
		filename = extract_filename(info_str, i)
	
		# Check if filename was found
		if filename:
	
			# Write the file
			with open("SCORES2/" + filename, "w") as f:
	
				f.write(gabc_download.read().decode("utf-8"))`

So maybe it would be nice for other people to have the data in a more convenient way, but I have the data I needed.

**Comment by igneus** [comment]
OK. Should anyone in future need the data, I can provide them.
And if I ever finish my toy project of an [alternative read-only frontend](https://github.com/igneus/gregoplus) over the GregoBase database, mass export of the scores will be part of it.

**Comment by jjbauer15** [comment]
@igneus Is there any update on this? Conversely, would read-only credentials to the database itself be on the table? Thank you!

**Comment by igneus** [comment]
@jjbauer15 If you are looking for a GregoBase export, last December another
guy asked me and I generated this http://yakub.cz/gregobase_export/ - gabc
of all scores (based on the GregoBase db dump available back then), grouped
by source and genre. Hope it helps.

(Should the code used for this be of interest, it's here
https://github.com/igneus/adiutor/commit/2895ed61f4f24c4ed3fd028a5deac3cc7c11c707
)

On the gregoplus app I referenced in my earlier comment there is no update.
It hasn't been and probably won't ever be launched, I have other priorities.

čt 16. 11. 2023 v 22:21 odesílatel jjbauer15 ***@***.***>
napsal:

**Comment by johnnysteen** [comment]
Would it be easy to update the gregobase_online.sql file with new changes made in the past few years? Or, rather than a text dump, a read-only dblink would be helpful.

**Comment by gigiodelchiaro** [comment]
@igneus I was scrolling through these issues and noticed your mention to your unfinished gregoplus app. I have made an app called GregoSearch, which also is a alternative UI for Gregobase. The only detail is that it is in Portuguese.

Just putting it out there in case you're interested.

If there is any update in this issue, I would be interested since I could have an up to date data base (my current one is from 2 yrs ago)

**Comment by juliantrue** [comment]
I was wrestling with the database dump file too. 
Decided to stand a up a repo for my work [here](https://github.com/juliantrue/GregoBaseExtract). Hopefully this can save you all some headache.

@igneus if you update the the database dump file on a semi regular basis, I can just run the extract job against my repo and keep things easy for you since it seems like desiring an extract of the database comes up with some frequency.

**Comment by igneus** [comment]
@juliantrue There seems to be some misunderstanding. I don't have access to the GregoBase database, therefore I can't dump it, and I don't manage the repository where it is made available either. All I ever did was in nature similar to what you've done - take the dump, import it in a database server, export some data.

**Comment by juliantrue** [comment]
@igneus ah my apologies. I just got here
